#
# old_revision [9fd950ff4de5331c13cd762cb755741046bdd576]
#
# add_file "libpurple/protocols/oscar/oscar-adium.c"
#  content [0dedf7511cd7c1ecbbdaff20bd38c401916b1be4]
# 
# add_file "libpurple/protocols/oscar/oscar-adium.h"
#  content [8237b3ee25499418b9a8118ad969f80195f736dc]
# 
# patch "libpurple/protocols/oscar/Makefile.am"
#  from [3dc744e59284b9c0d753237c89799db02a452015]
#    to [d280f99bdc4684680d098685bcde40e05a4761e3]
#
============================================================
--- libpurple/protocols/oscar/oscar-adium.c	0dedf7511cd7c1ecbbdaff20bd38c401916b1be4
+++ libpurple/protocols/oscar/oscar-adium.c	0dedf7511cd7c1ecbbdaff20bd38c401916b1be4
@@ -0,0 +1,17 @@
+#import "oscar-adium.h"
+
+void oscar_reformat_screenname(PurpleConnection *gc, const char *nick) {
+	OscarData *od = gc->proto_data;
+	if (!aim_sncmp(purple_account_get_username(purple_connection_get_account(gc)), nick)) {
+		if (!flap_connection_getbytype(od, SNAC_FAMILY_ADMIN)) {
+			od->setnick = TRUE;
+			od->newsn = g_strdup(nick);
+			aim_srv_requestnew(od, SNAC_FAMILY_ADMIN);
+		} else {
+			aim_admin_setnick(od, flap_connection_getbytype(od, SNAC_FAMILY_ADMIN), nick);
+		}
+	} else {
+		purple_notify_error(gc, NULL, _("The new formatting is invalid."),
+							_("Screen name formatting can change only capitalization and whitespace."));
+	}	
+}
============================================================
--- libpurple/protocols/oscar/oscar-adium.h	8237b3ee25499418b9a8118ad969f80195f736dc
+++ libpurple/protocols/oscar/oscar-adium.h	8237b3ee25499418b9a8118ad969f80195f736dc
@@ -0,0 +1,23 @@
+#include "internal.h"
+
+#include "account.h"
+#include "accountopt.h"
+#include "buddyicon.h"
+#include "cipher.h"
+#include "conversation.h"
+#include "core.h"
+#include "debug.h"
+#include "ft.h"
+#include "imgstore.h"
+#include "network.h"
+#include "notify.h"
+#include "privacy.h"
+#include "prpl.h"
+#include "proxy.h"
+#include "request.h"
+#include "util.h"
+#include "version.h"
+
+#include "oscar.h"
+
+void oscar_reformat_screenname(PurpleConnection *gc, const char *nick);
============================================================
--- libpurple/protocols/oscar/Makefile.am	3dc744e59284b9c0d753237c89799db02a452015
+++ libpurple/protocols/oscar/Makefile.am	d280f99bdc4684680d098685bcde40e05a4761e3
@@ -34,6 +34,8 @@ OSCARSOURCES = \
 	oft.c               \
 	oscar.c             \
 	oscar.h             \
+	oscar-adium.c       \
+	oscar-adium.h		\
 	oscarcommon.h       \
 	oscar_data.c        \
 	peer.c              \
